src/lem/tests/nivc_stutter.rs

```diff
@@ -0,0 +1,83 @@
+use pasta_curves::pallas::Scalar as Fr;
+
+use crate::{
+    coprocessor::test::DumbCoprocessor,
+    eval::lang::Lang,
+    lem::{
+        eval::{evaluate, make_cprocs_funcs_from_lang, make_eval_step_from_config, EvalConfig},
+        store::Store,
+        Tag,
+    },
+    state::user_sym,
+    tag::ExprTag,
+};
+
+#[test]
+fn test_nivc_stutter() {
+    let mut lang = Lang::<Fr, DumbCoprocessor<Fr>>::new();
+    let dumb = DumbCoprocessor::new();
+    let name = user_sym("cproc-dumb");
+
+    let store = Store::<Fr>::default();
+    lang.add_coprocessor(name, dumb);
+
+    let lurk_step = make_eval_step_from_config(&EvalConfig::new_nivc(&lang));
+    let cprocs = make_cprocs_funcs_from_lang(&lang);
+
+    assert_eq!(cprocs.len(), 1);
+    let cproc = &cprocs[0];
+
+    // 9^2 + 8 = 89
+    let expr = "(cproc-dumb 9 8)";
+    let expr = store.read_with_default_state(expr).unwrap();
+
+    let (frames, _) = evaluate(Some((&lurk_step, &lang)), expr, &store, 10).unwrap();
+
+    assert_eq!(frames.len(), 5);
+
+    // `cproc` can't reduce the first input
+    let first_input = &frames[0].input;
+    let output = &cproc
+        .call_simple(first_input, &store, &lang, 0)
+        .unwrap()
+        .output;
+    assert_eq!(first_input, output);
```

Comment:
I may not understand exactly what's going on here, but based on my reading of this test overall, it seems like this might indicate a test validating the wrong behavior.

A coprocessor should *not* 'stutter' on input it cannot reduce. It shouldn't produce a proof at all. That seems to be what the PR description suggests, but the assertion here looks equivalent to the following one which — I believe — is checking the Lurk stuttering behavior.

I think this should be 'unprovable' and therefore also (in this case) unevaluable.